src/ngrx/6-test-actions/test-actions.md

```diff
@@ -28,11 +28,85 @@ ng test --watch
+
+### Setting Up our TestBed
+
+When unit testing in general, we should use stubs to isolate parts of our application. Lucky NgRx makes this process simple by providing a way to create a mock `Store`:
+
+1. [MockStore](https://ngrx.io/api/store/testing/MockStore) - `MockStore` extends the `Store` class and [provides stubs for its methods](https://ngrx.io/guide/store/testing).
+
+2. [provideMockStore()](https://ngrx.io/api/store/testing/provideMockStore) - Generates a `MockStore` instance given a [configuration](https://ngrx.io/api/store/testing/MockStoreConfig).
+
+<details open>
+<summary>src/app/dashboard/dashboard.component.spec.ts</summary>
+@diff ../5-dispatch-actions/dashboard.component.spec.ts ./dashboard.component.spec-test-bed-setup.ts only
+</details>
+
+
+### Verify `LoginActions.logout` Action Dispatches Properly
+
+In `src/app/dashboard/dashboard.component.spec.ts`, there is a `describe` block with a `TODO`: "Spy on dispatching action". We will create a [spy](https://jasmine.github.io/tutorials/your_first_suite#:~:text=%C2%B6-,Spies,-Jasmine%20has%20test) to track when our `MockStore` dispatches an Action:
+
+<details open>
+<summary>src/app/dashboard/dashboard.component.spec.ts</summary>
+@diff ./dashboard.component.spec-test-bed-setup.ts ./dashboard.component.spec-spy-on-store.ts only
+</details>
+
+Next we will verify that the expected Action was dispatched after `logout()` is called:
 
-You'll need to copy the contents of two test files to run tests for your Actions. Note that we don't typically test Actions directly, we test their use (in Components, effects, reducers) but for the purposes of this lesson, we've included Action tests to make sure your code is in good shape!
+<details open>
+<summary>src/app/dashboard/dashboard.component.spec.ts</summary>
+@diff ./dashboard.component.spec-spy-on-store.ts ./dashboard.component.spec.ts only
+</details>
+
+> Note that there are 2 more pending `TODO`s in `src/app/dashboard/dashboard.component.spec.ts` that will be resolved in upcoming sections. For now we'll only be testing our Actions
+
+
+## Update `login.component.spec.ts`
+
+We will walk through updating `src/app/login/login.component.spec.ts` to run tests for your Actions.
+
+### Setting Up our TestBed
+
+<details open>
+<summary>src/app/login/login.component.spec.ts</summary>
+@diff ../5-dispatch-actions/login.component.spec.ts ./login.component.spec-test-bed-setup.ts only
+</details>
+
+
+### Verify `LoginActions.login` Action Dispatches With Form Payload Properly
+
+Unlike the `LoginActions.logout` Action, `LoginActions.login` Action requires a payload when dispatched. In our test, we will pass the state of our form:
+
+<details open>
+<summary>src/app/login/login.component.spec.ts</summary>
+@diff ./login.component.spec-test-bed-setup.ts ./login.component.spec-dispatch-action.ts only
+</details>
+
+
+### Verify `LoginActions.login` Action Does NOT Dispatches When Necessary
```

Comment:
does not dispatches (does not dispatch?)

Reply:
I think it's "does not dispatch", done~